lib/jekyll/converters/scss.rb

```diff
@@ -75,11 +75,28 @@ def sass_dir_relative_to_site_source
       end
 
       def sass_load_paths
+        paths = user_sass_load_paths + [sass_dir_relative_to_site_source]
+
         if safe?
-          [sass_dir_relative_to_site_source]
-        else
-          (user_sass_load_paths + [sass_dir_relative_to_site_source]).uniq
-        end.select { |load_path| File.directory?(load_path) }
+          # Sanitize paths to prevent any attack vectors (.e.g. `/**/*`)
+          paths.map! { |path| Jekyll.sanitized_path(@config["source"], path) }
```

Comment:
Additionally, `@config["source"]` comes from `site.config`. A quick solution is to [do what Site does](https://github.com/jekyll/jekyll/blob/803c8827e35ca9b9973427c4d7bf05f96ddf4348/lib/jekyll/site.rb#L21):

```ruby
@source = File.expand_path(@config["source"]).freeze
```